packages/odata/src/services/grid-odata.service.ts

```diff
@@ -426,8 +426,20 @@ export class GridOdataService implements BackendService {
             fieldName = titleCase(getHtmlStringOutput(fieldName || ''));
           }
 
-          // StartsWith + EndsWith combo
-          if (operator === OperatorType.startsWithEndsWith && Array.isArray(searchTerms) && searchTerms.length === 2) {
+          let predicateResult: string | undefined = undefined;
+          if (typeof this._odataService?.options?.filterPredicate === 'function') {
+            predicateResult = this._odataService.options.filterPredicate({
+              fieldName: getHtmlStringOutput(fieldName),
+              columnDef,
+              operator,
+              columnFilterOperator: columnFilter.operator,
+              searchValue
+            });
+          }
+
+          if (predicateResult !== undefined) {
+            searchBy = predicateResult;
```

Comment:
I'm not sure if it's necessary or not, but when I introduced `filterPredicate` for local JSON dataset, I was returning the filter predicate result all the time, however that had an indirect regression that I forgot to test at first and fixed it later. The regression was that when predicate returns `true`, it assumed it was true for all other column filters (which is obviously wrong), so I fixed the regression by only returning when the predicate result is `false`.. but I'm not sure if it's relevant in here though, I'm just adding this a step to verify

https://github.com/ghiscoding/slickgrid-universal/blob/a2799f2234eece619ec949165b76c23285c91698/packages/common/src/services/filter.service.ts#L367-L373